### ISSUE TITLE
Way to enable/disable a phase in EPM

### ISSUE THREAD

**Issue by Ank4n** [paritytech/substrate#13044]
During an ongoing incident, we might want to alter the phase in EPM in order to
- stop election by enabling Emergency Phase.
- reset normal election by rotating round, or switching phase to `off`.
- tell epm to be prepared for the next election manually by setting phase to `Signed` and then `forcing era` on staking once epm has a signed solution.

These ops could be very useful during a crisis. Specifically, we should have a call (or multiple calls) that allows `ForceOrigin` to transition EPM to a new phase.

**Comment by Szegoo** [comment]
I will open a PR for this.

**Comment by Ank4n** [comment]
@Szegoo Thanks for picking up so quickly. Its my bad, I didn't put the right labels. I need to confirm that we want to do this with @kianenigma @gpestana and others before we do this.

**Comment by Szegoo** [comment]
@Ank4n Did you get some feedback to see whether we want to do this?

**Comment by kianenigma** [comment]
#12972

**Comment by Ank4n** [comment]
> @Ank4n Did you get some feedback to see whether we want to do this?

@Szegoo Yes, we want to do this. Also have a look at the PR that Kian linked (https://github.com/paritytech/substrate/pull/12972) where these calls could be used to test more scenarios and how we can manage an incident in future.

**Comment by Szegoo** [comment]
> * tell EPM to be prepared for the next election manually by setting phase to `Signed` and then `forcing era` on staking once epm has a signed solution.

@Ank4n So we should have a storage item(configurable by force origin via an extrinsic) that would signal whether we want to force era on staking when the signed phase is finalized. Or did you think of a different approach?

**Comment by Ank4n** [comment]
> > * tell EPM to be prepared for the next election manually by setting phase to `Signed` and then `forcing era` on staking once epm has a signed solution.
> 
> @Ank4n So we should have a storage item(configurable by force origin via an extrinsic) that would signal whether we want to force era on staking when the signed phase is finalized. Or did you think of a different approach?

@Szegoo We already have a way to tell staking to force_era at next session change in the storage item `ForceEra`. Once we set `ForceEra` to `ForceNew`,  the EPM will start preparing for the next election in the next block. The issue with this approach though is, it might happen that the block at which `ForceNew` was set and the session change may be very few blocks apart or in the worst case, same block (which has already happened once) ending up in a situation where Staking asks EPM to do an election but EPM is not ready yet. So this approach is somewhat unreliable.

One approach could be that
- We first tell EPM to already start prepping for Election (i.e. go into `Signed Phase`).
- After at least `X` blocks, given `X` blocks are enough for EPM to be ready with a signed solution, we tell Staking to `ForceEra`. This way we can guarantee that EPM had enough time to be ready with a signed solution.

In a real scenario, above steps could potentially be one batch transaction to `governance`, with the extrinsic to `ForceEra` be a future scheduled operation to be executed at a given block number that we believe would be at least X blocks after the democracy action is executed. Though admittedly, this is still little tricky to do. Happy to explore if you have some other suggestions.

There is another scenario where telling EPM to go to the `Signed phase` would be useful is when EPM has gone into an emergency phase. In this case, Staking keeps asking for new election but since EPM is in emergency phase, it keeps failing. If we have a way to tell EPM to prepare for next election, the system has a change to recover to healthy state again. 

Eventually the things we do in this issue, we should be able to use them and demonstrate the scenarios above in the [Integration tests](https://github.com/paritytech/substrate/pull/12972).

**Comment by Szegoo** [comment]
@Ank4n Yes the second solution makes more sense.

AFAIU the EPM starts the signed phase in the last session of an era. This means that having an election that lasts a whole session is enough time to find a solution in most cases. When calling `force_new_era` we are saying that at the end of the current session, we want to start a new era, right? 

So why don't instead of waiting for X blocks to call the `force_new_era` function we instead at the beginning of a session start the `Signed phase` and also make a call to `force_new_era`? This means that the election will last the whole session which should be enough.

**Comment by Ank4n** [comment]
If we rely on timing a batch call then there is no issue since as soon as staking is told to force new era (and if we can do it at the start of a session), the EPM will start a Signed phase automatically. For this reason, my first suggestion also is a bit complicated. The better approach would be something that @gpestana was working on [here](https://github.com/paritytech/substrate/pull/13040). 

In regards to this PR though, we should just look into providing safe ways to steer EPM to a different phase than the default flow. We can experiment with various scenarios of things going wrong and how to recover from them in the PR related to the integration test.
